File: coindex/src/main/java/com/ironhack/coindex/model/Portfolio.java
package com.ironhack.coindex.model;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.Collection;

@Entity
public class Portfolio {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    @OneToMany(mappedBy = "portfolio", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private Collection<Position> positions;
    @ManyToOne
    private UserProfile userProfile;
    @CreationTimestamp
    private LocalDateTime creationDateTime;
    @UpdateTimestamp
    private LocalDateTime updateDateTime;


    public Portfolio() {
    }

    public Portfolio(String name, String description, Collection<Position> positions, UserProfile userProfile) {
        this.name = name;
        this.description = description;
        this.positions = positions;
        this.userProfile = userProfile;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Collection<Position> getPositions() {
        return positions;
    }

    public void setPositions(Collection<Position> positions) {
        this.positions = positions;
    }

    public UserProfile getUserProfile() {
        return userProfile;
    }

    public void setUserProfile(UserProfile userProfile) {
        this.userProfile = userProfile;
    }



    public LocalDateTime getUpdateDateTime() {
        return updateDateTime;
    }

    public void setUpdateDateTime(LocalDateTime updateDateTime) {
        this.updateDateTime = updateDateTime;
    }

    public LocalDateTime getCreationDateTime() {
        return creationDateTime;
    }

    public void setCreationDateTime(LocalDateTime creationDateTime) {
        this.creationDateTime = creationDateTime;
    }

    public boolean addPosition(Position position) {
        return this.getPositions().add(position);
    }

}

File: frontend/src/app/models/portfolio.ts
import {Position } from './position'

export class Portfolio {

    // private Long id;
    // private String name;
    // private String description;
    // private List<PositionDTO> positions;
    // private Long idUserProfile;
    constructor(
        private _id: number,
        private _name: string,
        private _description: string,
        private _idUserProfile: number,
        private _positions: Position[]
    ) {  
    }

    public get id(): number {
        return this._id;
    }
    
    public get name(): string {
        return this._name;
    }

    public get description(): string {
        return this._description;
    }

    public get idUserProfile(): number {
        return this._idUserProfile;
    }

    public get positions(): Position[] {
        return this._positions;
    }

    public set id(id: number) {
        this._id = id;
    }

    public set name(name: string) {
        this._name = name;
    }

    public set description(description: string) {
        this._description = description;
    }

    public set idUserProfile(idUserProfile: number) {
        this._idUserProfile = idUserProfile;
    }

    public set positions(positions: Position[]) {
        this._positions = positions;
    }
    
}


